scalafix-core/shared/src/main/scala/scalafix/internal/rule/Disable.scala

```diff
@@ -1,18 +1,88 @@
 package scalafix.internal.rule
 
 import metaconfig.{Conf, Configured}
+
 import scala.meta._
-import scalafix.internal.config.DisableConfig
-import scalafix.lint.LintCategory
-import scalafix.lint.LintMessage
-import scalafix.rule.SemanticRule
-import scalafix.rule.{Rule, RuleCtx}
+import scala.meta.transversers.Traverser
+import scalafix.internal.config.{DisableConfig, DisabledSymbol}
+import scalafix.internal.util.SymbolOps
+import scalafix.lint.{LintCategory, LintMessage}
+import scalafix.rule.{Rule, RuleCtx, SemanticRule}
 import scalafix.util.SemanticdbIndex
-import scalafix.util.SymbolMatcher
+
+object Disable {
+
+  /**
+    * A tree traverser to collect values with a custom context.
+    * At every tree node, either builds a new Context or returns a new Value to accumulate.
+    * To collect all accumulated values, use result(Tree).
+    */
+  class ContextTraverser[Value, Context](initContext: Context)(
+      fn: PartialFunction[(Tree, Context), Either[Value, Context]])
+      extends Traverser {
+    private var context: Context = initContext
+    private val buf = scala.collection.mutable.ListBuffer[Value]()
+
+    private val liftedFn = fn.lift
+
+    override def apply(tree: Tree): Unit = {
+      liftedFn((tree, context)) match {
+        case Some(Left(res)) =>
+          buf += res
+        case Some(Right(newContext)) =>
+          val oldContext = context
+          context = newContext
+          super.apply(tree)
+          context = oldContext
+        case None =>
+          super.apply(tree)
+      }
+    }
+
+    def result(tree: Tree): List[Value] = {
+      context = initContext
+      buf.clear()
+      apply(tree)
+      buf.toList
+    }
+  }
+
+  def matchDisabledSymbol(disabledSymbol: DisabledSymbol, symbol: Symbol)(
+      implicit index: SemanticdbIndex): Boolean =
+    disabledSymbol match {
+      case DisabledSymbol(Some(s), _, _, banHierarchy, _) =>
+        SymbolOps.isSameNormalized(symbol, s) || (
+          banHierarchy &&
+            index
+              .denotation(symbol)
+              .exists(_.overrides.exists(SymbolOps.isSameNormalized(symbol, _)))
```

Comment:
Here, `ovverides` is empty list.

Reply:
I suspect it's because you need to add a `scalacOption`

https://github.com/MasseGuillaume/scalafix/blob/09666ec5406759737b80c4ef0abc5252c2f556e0/build.sbt#L263-L273

Reply:
@MasseGuillaume It doesn't changes anything...

Reply:
iirc, you need both
```
-P:semanticdb:overrides:all
-P:semanticdb:denotations:all
```

However, denotations:all should no longer be necessary soon thanks to metacp. I think it's best to wait with depending on `overrides` for now